### PR TITLE
Fix incorrect RBAC permissions in object creation tasks

### PR DESCRIPTION
**Pull request by gerrod3**
fixes: #5683

### CODE REVIEW

File: pulpcore/tasking/_util.py [hunk 1]
```diff
@@ -97,7 +97,26 @@ def perform_task(task_pk, task_working_dir_rel_path):
     connection.connection = None
     # enc_args and enc_kwargs are deferred by default but we actually want them
     task = Task.objects.defer(None).select_related("pulp_domain").get(pk=task_pk)
-    user = get_users_with_perms(task, with_group_users=False).first()
+    # These queries were specifically constructed and ordered this way to ensure we have the highest
+    # chance of getting the user who dispatched the task since we don't have a user relation on the
+    # task model. The second query acts as a fallback to provide ZDU support. Future changes will
+    # require to keep these around till a non-ZDU release is planned (probably 3.70 the earliest).
```
Comment (mdellweg) [inline pulpcore/tasking/_util.py:103]: Technically, if we get it right, the breaking change releases still allow upgrades without downtime.
```suggestion
    # require to keep these around till a breaking change release is planned (probably 3.70 the earliest).
```

**Review by mdellweg — APPROVED** [review]
One suggestion to the comment if you want to... Approving anyway.

**Comment by mdellweg** [comment]
Can you add a reference to the CVE in the changelog and the commit message please?

**Review by mdellweg — APPROVED** [review]

**Review by ipanova — APPROVED** [review]

**Comment by mdellweg** [comment]
I'm going to care for the manual backports now... *sigh*

**Pull request merged**